src/tracer.rs

```diff
@@ -4,20 +4,31 @@ pub static LOG_CALL: &str = "__log_call";
 pub static EXPOSE_TRACER: &str = "__expose_tracer";
 pub static EXPOSE_TRACER_LEN: &str = "__expose_tracer_len";
 
+static RING_BUFFER_ENTRIES: usize = 1024;
+
+#[repr(i32)]
+#[derive(Debug)]
+pub enum EntryKind {
+    FunctionCall = 0,
+    FunctionReturnVoid = 1,
+    FunctionReturnValue = 2,
+}
+
 /// Wrapper around the ring buffer for recording function calls.
 #[derive(Debug)]
-pub struct Tracer(RingBuffer<u32>);
+pub struct Tracer(RingBuffer<i32>);
 
 impl Tracer {
     pub fn new() -> Self {
-        Tracer(RingBuffer::new(1024))
+        Tracer(RingBuffer::new(RING_BUFFER_ENTRIES * 2))
     }
 
-    pub fn log(&mut self, data: u32) {
+    pub fn log(&mut self, kind: i32, data: i32) {
+        self.0.enqueue(kind as i32);
```

Comment:
Do we want to support enqueuing pairs or find a way to show the relationship between the `kind` & `data` in the buffer?